### ISSUE TITLE
Depricate VTI interfaces for the 5.0 release?

### ISSUE THREAD

**Issue by bradyallenjohnson** [libreswan/libreswan#1169]
In yesterday's (June 14) team meeting, we were discussing how to proceed with my patches to refcount XFRMi and VTI interface and IP addresses on those interfaces. It was mentioned in the meeting that Libreswan would like to stop supporting VTI interfaces and perhaps we can do that once and for all in the forthcoming 5.0 release.

I am creating this issue to start a discussion on this and hopefully reach an agreement.

**Comment by letoams** [comment]
On Thu, 15 Jun 2023, Brady Johnson wrote:

> In yesterday's (June 14) team meeting, we were discussing how to proceed with my patches to refcount XFRMi and VTI interface and IP
> addresses on those interfaces. It was mentioned in the meeting that Libreswan would like to stop supporting VTI interfaces and
> perhaps we can do that once and for all in the forthcoming 5.0 release.
> 
> I am creating this issue to start a discussion on this and hopefully reach an agreement.

There are use cases people currently use with VTI that are not supported
with XFRMi. I'm not sure if I am ready to drop VTI support right now.

**Comment by cagney** [comment]
> There are use cases people currently use with VTI that are not supported with XFRMi.

Which usecases specifically?

**Comment by letoams** [comment]
On Thu, 15 Jun 2023, cagney wrote:

>       There are use cases people currently use with VTI that are not supported with XFRMi.
> 
> Which usecases specifically?

I don't remember exactly or I would have mentioned it here. Possible
transport mode or some 0.0.0.0/0 use cases?

**Comment by paulwouters** [comment]
I don't think that is correct, but I would have to go down hunting for emails/discussions about the use cases

**Comment by cagney** [comment]
punt to 5.1
